### ISSUE TITLE
No validation set detected with validation_split:0

### ISSUE THREAD

**Issue by Philip642** [jgraving/DeepPoseKit#43]
Hi Jake,

I have now  over 3h video footage from different animals doing different behaviors and I got the message No validation set detected with validation_split:0 so I changed it to 0.1.

Fit the model works  with these settings:
model.fit(
    batch_size=12, #less than n_train from data generator
    validation_batch_size=1, # less than n_val from data generator
    callbacks=callbacks,
    #epochs=1000, # Increase the number of epochs to train the model longer
    epochs=200,
    n_workers=1,
    steps_per_epoch=500,
)

and the output from the training generator looks like this:
{'n_train': 19,
 'n_validation': 2,
 'validation_split': 0.1,
 'downsample_factor': 3,
 'output_shape': (48, 48),
 'n_output_channels': 88,
 'shuffle': True,
 'sigma': 5,
 'output_sigma': 0.625,
 'use_graph': True,
 'graph_scale': 1,
 'random_seed': 1,
 'augmenter': True,
 'datapath': '/home/einat/deepposekit-data/datasets/whipspider/annotation_data_release.h5',
 'dataset': 'images',
 'generator': 'DataGenerator',
 'n_samples': 21,
 'image_shape': (384, 384, 1),
 'keypoints_shape': (43, 2)}

Output from fit the model
Epoch 1/200
 13/500 [..............................] - ETA: 2:36:46 - loss: 178.4359 - output_0_loss: 71.6774 - output_1_loss: 56.4588 - output_2_loss: 50.2997

Epoch 1/200
405/500 [=======================>......] - ETA: 29:31 - loss: 52.7862 - output_0_loss: 19.2698 - output_1_loss: 17.0771 - output_2_loss: 16.4393

Does this looks good?

Cheers, Philip

**Comment by jgraving** [comment]
Yes, if you set validation split to 0 then there will be no validation set. If the loss is decreasing then training is working. You should assess the quality of the results using the notebooks for step 4.
